lib/fluent/plugin/in_system_metrics.rb

```
@@ -0,0 +1,143 @@
+            
+            metrics['memory.total_bytes'] = total
+            metrics['memory.free_bytes'] = free
+            metrics['memory.available_bytes'] = available
+            metrics['memory.used_bytes'] = total - free - buffers - cached
+            metrics['memory.usage_percent'] = 100.0 * (total - available) / total if total > 0
+          end
+        rescue => e
+          log.error "Failed to collect memory metrics", error: e
+        end
+        metrics
+      end
```

RuboCop (1.75.5)

[convention] Assignment Branch Condition size for collect_memory_metrics is too high. [<13, 35, 5> 37.67/23]

(Metrics/AbcSize)